src/nifreeze/data/dmri.py

```diff
@@ -71,12 +71,43 @@ class DWI(BaseDataset[np.ndarray]):
     bzero: np.ndarray = attrs.field(default=None, repr=_data_repr, eq=attrs.cmp_using(eq=_cmp))
     """A *b=0* reference map, preferably obtained by some smart averaging."""
     gradients: np.ndarray = attrs.field(default=None, repr=_data_repr, eq=attrs.cmp_using(eq=_cmp))
-    """A 2D numpy array of the gradient table (4xN)."""
+    """A 2D numpy array of the gradient table (``N`` orientations x ``C`` components)."""
     eddy_xfms: list = attrs.field(default=None)
     """List of transforms to correct for estimated eddy current distortions."""
 
+    def __attrs_post_init__(self) -> None:
+        self._normalize_gradients()
+
+    def _normalize_gradients(self) -> None:
+        if self.gradients is None:
+            return
+
+        gradients = np.asarray(self.gradients)
+        if gradients.ndim != 2:
+            raise ValueError("Gradient table must be a 2D array")
+
+        n_volumes = None
+        if self.dataobj is not None:
```

Comment:
@jhlegarreta I think we disallowed this right (dataobj to be None)?

Reply:
Not fully yet: see #302 and PR #305.